### DISCUSSION TITLE
WinUI3: How to hide the mouse cursor

### DISCUSSION THREAD

**Discussion by AliGuemues** [microsoft/microsoft-ui-xaml#7872] [Q&A]
Hello,

I am developing a WinUI3 application and want to hide the mouse cursor after some seconds of inactivity, or generally want to hide the cursor in some cases.

I searched a lot on the internet but didn´t find anything helpful for my case. Is it possible to achieve something like this?

**Comment by castorix** [comment]
As [ShowCursor ](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showcursor) only works for Win32 controls,
a way is with _InputDesktopResourceCursor.CreateFromModule_ and a custom empty cursor in a DLL
then _set_ProtectedCursor_ with Reflection

**Reply by AndrewKeepCoding** [reply]
> As [ShowCursor ](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showcursor) only works for Win32 controls, a way is with _InputDesktopResourceCursor.CreateFromModule_ and a custom empty cursor in a DLL then _set_ProtectedCursor_ with Reflection

Is this supposed to work on ``Page``s?

I'm trying to hide the cursor using the code below, but it only hides it only when it's on a control (``Button``s, ``ColorPicker``s, etc).

```cs
private void Button_Click(object sender, RoutedEventArgs e)
{
    // Get an empty cursor from resources.
    InputCursor emptyCursor = GetEmptyCursor();

    // Call this method using Reflection.
    // IUIElementProtectedMethods.set_ProtectedCursor(_objRef_global__Microsoft_UI_Xaml_IUIElementProtected, value);
    if (Assembly.GetAssembly(typeof(UIElement)) is Assembly assembly &&
        assembly.GetType("ABI.Microsoft.UI.Xaml.IUIElementProtectedMethods") is Type targetType &&
        targetType.GetMethod(
            "set_ProtectedCursor",
            BindingFlags.Static | BindingFlags.Public | BindingFlags.Instance) is MethodInfo methodInfo &&
        typeof(UIElement).GetProperty(
            "_objRef_global__Microsoft_UI_Xaml_IUIElementProtected",
            BindingFlags.NonPublic | BindingFlags.Instance) is PropertyInfo propertyInfo &&
            propertyInfo.GetValue(this) is IObjectReference objectReference)
    {
        methodInfo.Invoke(this, new object[] { objectReference, emptyCursor });
    }
}
```

**Comment by AliGuemues** [comment]
Thank you for your response. I will try that.
